cdc/changefeed.go

```diff
@@ -803,7 +816,41 @@ func (c *changeFeed) calcResolvedTs(ctx context.Context) error {
 
 	var tsUpdated bool
 
-	if minResolvedTs > c.status.ResolvedTs {
+	//syncpoint on
+	if c.info.SyncPointEnabled {
+		c.syncpointMutex.Lock()
+		defer c.syncpointMutex.Unlock()
+		if c.status.ResolvedTs == c.status.CheckpointTs && !c.updateResolvedTs && c.ddlTs == 0 {
```

Comment:
The logic in L823 and L837 are complicated, recommend adding comments to describe the detailed logic.